src/core_modules/capture-core/components/WidgetEnrollment/HOC/withTrackedEntityInstances.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import React, { useMemo, type ComponentType } from 'react';
+import { useDataQuery } from '@dhis2/app-runtime';
+
+type Props = {
+    programId: string,
+    teiId: string,
+};
+
+export const withTrackedEntityInstances = (Component: ComponentType<any>) => (
+    props: Props,
+) => {
+    const { error, loading, data } = useDataQuery(
+        useMemo(
+            () => ({
+                trackedEntityInstances: {
+                    resource: `trackedEntityInstances/${props.teiId}`,
+                    params: {
+                        fields: ['programOwners[ownerOrgUnit]'],
+                        program: [props.programId],
+                    },
+                },
+            }),
+            [props.teiId, props.programId],
+        ),
+    );
+
+    if (error) {
+        throw error;
+    }
+
+    return !loading &&
```

Comment:
this condition is quite complex, I would suggest separate them and using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).